modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/AccessTokenHandler.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.mdm.agent.proxy;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+import android.os.AsyncTask;
+import android.util.Log;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.mdm.agent.proxy.beans.EndPointInfo;
+import org.wso2.mdm.agent.proxy.beans.CredentialInfo;
+import org.wso2.mdm.agent.proxy.beans.Token;
+import org.wso2.mdm.agent.proxy.interfaces.CallBack;
+import org.wso2.mdm.agent.proxy.utils.Constants;
+import org.wso2.mdm.agent.proxy.utils.ServerUtilities;
+import org.wso2.mdm.agent.proxy.utils.Constants.HTTP_METHODS;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * After receiving authorization code client application can use this class to
+ * obtain access token
+ */
+public class AccessTokenHandler extends Activity {
+	private static final String TAG = "AccessTokenHandler";
+	private static final String USERNAME_LABEL = "username";
+	private static final String PASSWORD_LABEL = "password";
+	private static final String COLON = ":";
+	private static final DateFormat dateFormat =
+			new SimpleDateFormat("MM/dd/yyyy HH:mm:ss", Locale.getDefault());
+	private CredentialInfo info;
+
+	public AccessTokenHandler(CredentialInfo info, CallBack callBack) {
+		this.info = info;
+	}
+
+	public void obtainAccessToken() {
+		new NetworkCallTask().execute();
```

Comment:
Rather than creating ad-hoc instances of NetworkClassTask, check if we can possibly use some sort of a pooling mechanism to control the creation of NetworkClassTask instances.